### PR TITLE
discoveryfeed: Rotate hooks by day, not randomly

### PR DESCRIPTION
**Pull request by smspillaz**
https://phabricator.endlessm.com/T16982

### CODE REVIEW

File: search-provider/eks-discovery-feed-provider.c [hunk 1]
```diff
@@ -203,11 +203,27 @@ add_key_value_pair_from_model_to_variant (EkncContentObjectModel *model,
   add_key_value_pair_to_variant (builder, underscore_key, value);
 }
 
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(GDate, g_date_free);
```
Comment (cosimoc) [inline search-provider/eks-discovery-feed-provider.c]: Remove terminal semicolon (but you may not need this)

File: search-provider/eks-discovery-feed-provider.c [hunk 2]
```diff
@@ -203,11 +203,27 @@ add_key_value_pair_from_model_to_variant (EkncContentObjectModel *model,
...
+
+static GDateWeekday
+get_day_of_week ()
```
Comment (cosimoc) [inline search-provider/eks-discovery-feed-provider.c]: The function parameters should be `(void)`

File: search-provider/eks-discovery-feed-provider.c [hunk 3]
```diff
@@ -203,11 +203,27 @@ add_key_value_pair_from_model_to_variant (EkncContentObjectModel *model,
...
+{
+  GTimeVal tv;
+  g_autoptr(GDate) date = g_date_new ();
```
Comment (cosimoc) [inline search-provider/eks-discovery-feed-provider.c]: Why not use `GDateTime` instead? I think it has simpler API for this.

```
g_autoptr(GDateTime) datetime = g_date_time_new_now_local();
return g_date_time_get_day_of_week (datetime);
```

Comment (smspillaz) [inline search-provider/eks-discovery-feed-provider.c]: Yep, I was looking for that. Thanks!

File: search-provider/eks-discovery-feed-provider.c [hunk 4]
```diff
@@ -203,11 +203,18 @@ add_key_value_pair_from_model_to_variant (EkncContentObjectModel *model,
   add_key_value_pair_to_variant (builder, underscore_key, value);
 }
 
+static GDateWeekday
+get_day_of_week (void)
+{
+  g_autoptr(GDateTime) datetime = g_date_time_new_now_local ();
+  return g_date_time_get_day_of_week (datetime);
```
Comment (cosimoc) [inline search-provider/eks-discovery-feed-provider.c]: This returns a `gint`, not a `GDateWeekday`.

Comment (smspillaz) [inline search-provider/eks-discovery-feed-provider.c]: Yep, fair enough

**Comment by cosimoc** [comment]
@smspillaz Please rebase this to master.

**Comment by smspillaz** [comment]
@cosimoc done

**Pull request merged**